CaseStudyProductListing/Shared/NetworkManager/NetworkManager.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  NetworkManager.swift
+//  CaseStudyProductListing
+//
+//  Created by Atul Vishnoi on 13/07/23.
+//
+
+import Foundation
+
+enum NetworkError: Error {
+    case invalidUrl
+    case invalidData
+}
+
+enum FeedType: String {
+    case emailed
+    case shared
+    case viewed
+}
+
+struct NetworkManager {
+    // Base API URL
+    private let baseUrl = "https://api.nytimes.com/svc/mostpopular/v2/"
+    // API Key
+    private let apiKey = "RAkAhPXo0kaO0yqbS8W6ZfshjhjvZft8"
```

Comment:
API Key Security:
The API key is hardcoded in the code, which can be a security concern.
Storing the API key securely or retrieving it from a configuration file is recommended to protect sensitive information.